documents/alembic_migrations.md

```diff
@@ -0,0 +1,180 @@
+
+```text
+POSTGRES_USER=your_postgres_user
+POSTGRES_PASSWORD=your_postgres_password
+POSTGRES_DB=your_database_name
+
+DATABASE_URL_SYNC=postgresql://your_postgres_user:your_postgres_password@localhost/your_database_name
+DATABASE_URL_ASYNC=postgresql+asyncpg://your_postgres_user:your_postgres_password@localhost/your_database_name
+```
+
+* Replace `your_postgres_user`, `your_postgres_password`, and `your_database_name` with the credentials you choose.
+* These values will be used by Docker and Alembic to connect to the PostgreSQL database.
+
+---
+
+* **Docker Build and Start-up**
+Build Docker Containers: Use the following command to build Docker containers without using cached versions:
+
+```bash
+docker-compose build --no-cache
+```
+This command ensures that all components, including dependencies, are freshly built.
+
+2. Start Containers in Detached Mode:
+
+```bash
+docker-compose up -d
+```
+
+This initializes the containers in the background. The configuration specified in docker-compose.yml will set up both the database and the application, with the database container configured to include PostGIS extensions.
+
+## Database Initialization with init_postgis.sql
+
+The `init_postgis.sql` script is used for setting up the PostGIS environment in the PostgreSQL database. Here’s a breakdown of what this script does:
+
+---
+
+### Notes about the init_postgis.sql script executed in the Dockerfile
+
+(don't follow these steps, they are already executed in the Dockerfile)
+
+```sql
+CREATE EXTENSION IF NOT EXISTS plpgsql;
+```
+This command enables PostgreSQL's procedural language, allowing functions written in PL/pgSQL, a core language for stored procedures.
+
+**Install Additional Extensions:**
+
+* **fuzzystrmatch:** Provides functions for approximate string matching, useful for text comparisons.
+
+* **postgis:** Installs PostGIS, which extends PostgreSQL to handle geographic objects, making it suitable for geospatial analysis.
+
+* **postgis_raster:** Adds support for raster data (gridded data) in PostGIS.
+
+```sql
+CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;
+CREATE EXTENSION IF NOT EXISTS postgis;
+CREATE EXTENSION IF NOT EXISTS postgis_raster;
+```
+
+* **Enable GDAL Drivers and External Raster Support:**
+
+```sql
+DO $$
+BEGIN
+    EXECUTE format('ALTER DATABASE %I SET postgis.gdal_enabled_drivers = ''ENABLE_ALL''', current_database());
+    EXECUTE format('ALTER DATABASE %I SET postgis.enable_outdb_rasters = true', current_database());
+END $$;
+```
+
+These commands configure the database to use GDAL drivers (libraries for geospatial data) and enable storage of raster files outside the database, optimizing for large datasets.
+
+--- 
+
+## Step 2: Setting Up Alembic for Migrations
+
+1. **Initialize Alembic:**
+
+* In your directory you must have an alembic folder, <span style="color:red">if you don't have it </span>, you can create it with the following command:
+
+```bash
+alembic init
```

Comment:
I still encounter the same error when executing this command. It's now clear that it's unnecessary because the alembic directory already exists. However, it's important that the command works. 

Is a parameter named `directory` missing?

![imagen](https://github.com/user-attachments/assets/394cd5a6-7284-4edb-8970-0418fca6fc86)

Reply:
@ManuelStardust you have the same error?